### PR TITLE
fix(shared): fix dot validation

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

**Review by niklasnatter — APPROVED** [review]

File: packages/shared/src/broker.ts [hunk 1]
```diff
@@ -82,12 +89,6 @@ const validateRequest = (network: ChainflipNetwork, params: unknown) => {
     .superRefine(ensureDcaWithFok)
     .parse(params);
 
-  if (parsed.destAsset.chain === 'Polkadot') {
```
Comment (niklasnatter) [inline packages/shared/src/broker.ts:85]: we sure that we can remove this? 😬

Comment (acdibble) [reply packages/shared/src/broker.ts:85]: yeas, a new case has been added to the union where we transform the ss58 address into hex

**Review by jerryafr — APPROVED** [review]

File: packages/shared/src/parsers.ts [hunk 2]
```diff
@@ -91,14 +93,19 @@ export const ethereumAddress = hexString.refine(
...
+  (address) => ss58.decode(address).ss58Format === chainflipSS58Prefix,
   (address) => ({ message: `${address} is not a valid Chainflip address` }),
 );
 
 export const solanaAddress = string.refine(isValidSolanaAddress, (address) => ({
   message: `${address} is not a valid Solana address`,
 }));
 
+export const polkadotAddress = string.refine(
+  (address) => ss58.decode(address),
```
Comment (zoheb391) [inline packages/shared/src/parsers.ts:105]: still need to limit length to 66

Comment (acdibble) [reply packages/shared/src/parsers.ts:105]: `decode` checks that the address is 32 bytes. `hexString` doesn't have any limitation, which means that we don't check that EVM addresses either, i can add it for EVM too

Comment (zoheb391) [reply packages/shared/src/parsers.ts:105]: ah didn't realise decode checked it 👍

**Review by zoheb391 — APPROVED** [review]

**Review by GabrielBuragev — APPROVED** [review]

**Pull request merged**